spec/unit/ssl/host_spec.rb

```diff
@@ -767,14 +766,63 @@ def base_json_comparison(result, json_hash)
 
       it "retrieves it from the server" do
         Puppet::Rest::Routes.expects(:get_crls)
-          .with(@http, Puppet::SSL::CA_NAME)
+          .with(@http, Puppet::SSL::CA_NAME, nil)
           .yields(@pki[:crl_chain])
 
         @host.ssl_store
         expect(Puppet::FileSystem.read(Puppet.settings[:hostcrl], :encoding => Encoding::UTF_8)).to eq(@pki[:crl_chain])
       end
     end
 
+    context "and the CRL is on disk" do
+      before do
+        @pki = PuppetSpec::SSL.create_chained_pki
+        localcacert = Puppet.settings[:localcacert]
+        hostcrl     = Puppet.settings[:hostcrl]
+        Puppet::Util.replace_file(localcacert, 0644) {|f| f.write @pki[:ca_bundle] }
+        Puppet::Util.replace_file(hostcrl, 0644)     {|f| f.write @pki[:crl_chain] }
+        @http = mock 'http'
+        @host.stubs(:http_client).returns(@http)
+      end
+
+      after do
+        Puppet::FileSystem.unlink(Puppet.settings[:localcacert])
+        Puppet::FileSystem.unlink(Puppet.settings[:hostcrl])
+      end
+
+      context "and there is a newer crl available from the master" do
+        it "retrieves the new crl from the server and replaces the crl on disk" do
+          pki2 = PuppetSpec::SSL.create_chained_pki
+          crl_last_modified = File.mtime(Puppet.settings[:hostcrl])
+          Puppet::Rest::Routes.expects(:get_crls)
+            .with(@http, Puppet::SSL::CA_NAME, crl_last_modified)
+            .yields(pki2[:crl_chain])
+
+          @host.ssl_store
+          expect(Puppet::FileSystem.read(Puppet.settings[:hostcrl], :encoding => Encoding::UTF_8)).to eq(pki2[:crl_chain])
+        end
+      end
+
+      context "and there is NO newer crl available on the master" do
+        before (:each) do
+          crl_last_modified = File.mtime(Puppet.settings[:hostcrl])
+          Puppet::Rest::Routes.expects(:get_crls)
+            .with(@http, Puppet::SSL::CA_NAME, crl_last_modified)
+            .raises(Puppet::Rest::ResponseError.new('no crl returned',
+                                                    mock('response', status_code: 304)))
+        end
+
+        it "does not replace the crl on disk" do
+          @host.ssl_store
+          expect(Puppet::FileSystem.read(Puppet.settings[:hostcrl], :encoding => Encoding::UTF_8)).to eq(@pki[:crl_chain])
+        end
+
+        it "does not raise a puppet error" do
```

Comment:
I think that this could be combined with the test above it.

Reply:
How would you combine them?  I'm not sure how to do that since the tests require diff formats ( i mean one expects something after host.ssl_store is called and one is the block style).

Reply:
I think that these could be combined with a two `expect` lines within the it block. Does this work?

``` ruby
it 'does not raise an error and does not replace the crl on disk' do
  expect { @host.ssl_store }.not_to raise_error
  expect(Puppet::FileSystem.read(Puppet.settings[:hostcrl], :encoding => 
    Encoding::UTF_8)).to eq(@pki[:crl_chain])
end

```

Reply:
Yup! Thanks.  Fixed it.